providers/cleanup_checks.rb

```diff
@@ -0,0 +1,29 @@
+action :run do
+  Chef::Log.info("#{new_resource} Disabling up unmanaged Sensu checks")
+
+  definition_directory = ::File.join(node.sensu.directory, "conf.d", "checks")
+  @definition_path = ::File.join(definition_directory, "#{new_resource.name}.json")
+
+  file_list = ::Dir.glob(::File.join(definition_directory, '*.json'))
+
+  protected_files = run_context.resource_collection.all_resources.select do |resource|
+    resource.resource_name == "#{new_resource.cookbook_name}_check".to_sym
+  end.map do |resource|
+    ::File.join(definition_directory, "#{resource.name}.json")
+  end
+
+  checks_to_remove = file_list - protected_files
+
+  checks_to_remove.each do |file|
+    Chef::Log.info("Sensu check '#{file}' is not managed by Chef. Disabling it.")
+    ::File.unlink file
+  end
+
+  if not kill_them_with_fire.empty?
```

Comment:
Comparing with #288, looks like `kill_them_with_fire`should be updated to `checks_to_remove`?